Clinical-Genomics/scout
# Chanjo2 report for cases with d4 files

This PR adds a functionality or fixes a bug.
- Create chanjo2 reports for cases with associated d4 files (partial fix for #4000)
- To be tested in a combo with chanjo2, branch [speedup_report_alt](https://github.com/Clinical-Genomics/chanjo2/pull/236)

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test at CG**:
1. Deploy on cg-vm1
1. Go to cust000, case [safeguinea](https://scout-stage.scilifelab.se/cust000/F0006358-mip7)
1. Create coverage report by clicking on the relative link on the case side panel

**How to test on another server**:
1. Given an instance of chanjo2 that has access to the d4 file path associated to the scout case
1. You might need to install the async lib if you test this locally. pip install -r requirements.txt
1. Have ready one case with individuals having d4 files associated (see #4363)
1. Uncomment the scout setup config to point to your chanjo2 instance
1. Go to the case page and click on the coverage report on the side bar:

<img width="220" alt="image" src="https://github.com/Clinical-Genomics/scout/assets/28093618/63a8ff5e-7673-4d0e-88d7-6e45e2c694af">


**Expected outcome**:
- Chanjo2 report should be shown

**Review:**
- [x] code approved by Jakob37
